phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/mt/tenantoperation/TenantOperationFactory.java

```diff
@@ -0,0 +1,501 @@
+                    if (!operation.getPreScenarioDdls().isEmpty()) {
+                        try (Connection conn = phoenixUtil.getConnection(tenantId)) {
+                            for (Ddl ddl : scenario.getPreScenarioDdls()) {
+                                LOGGER.info("\nExecuting DDL:" + ddl + " on tenantId:" + tenantId);
+                                phoenixUtil.executeStatement(ddl.toString(), conn);
+                                if (ddl.getStatement().toUpperCase().contains(phoenixUtil.ASYNC_KEYWORD)) {
+                                    phoenixUtil.waitForAsyncIndexToFinish(ddl.getTableName());
+                                }
+                            }
+                        } catch (SQLException throwables) {
+                            throw new RuntimeException(throwables);
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    long totalDuration = EnvironmentEdgeManager.currentTimeMillis() - startTime;
+                    return new OperationStats(input, startTime,0, operation.getPreScenarioDdls().size(), totalDuration);
+
+                }
+            };
+        }
+    }
+
+    @VisibleForTesting
+    class NoopTenantOperationImpl implements TenantOperationImpl {
+
+        @Override public Function<TenantOperationInfo, OperationStats> getMethod() {
+            return new Function<TenantOperationInfo, OperationStats>() {
+                @Override public OperationStats apply(final TenantOperationInfo input) {
+
+                    final NoopOperation operation = (NoopOperation) input.getOperation();
+                    final Noop noop = operation.getNoop();
+
+                    long startTime = EnvironmentEdgeManager.currentTimeMillis();
+                    // Sleep for the specified time to simulate idle time.
+                    try {
+                        TimeUnit.MILLISECONDS.sleep(noop.getIdleTime());
+                        long duration = EnvironmentEdgeManager.currentTimeMillis() - startTime;
+                        return new OperationStats(input, startTime, 0, 0, duration);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                        long duration = EnvironmentEdgeManager.currentTimeMillis() - startTime;
+                        return new OperationStats(input, startTime,-1, 0, duration);
+                    }
+                }
+            };
+        }
+    }
+
+    class UserDefinedOperationImpl implements TenantOperationImpl {
+
+        @Override public Function<TenantOperationInfo, OperationStats> getMethod() {
+            return new Function<TenantOperationInfo, OperationStats>() {
+                @Override public OperationStats apply(final TenantOperationInfo input) {
+                    // TODO : implement user defined operation invocation.
```

Comment:
Can you create a Jira for this and link that in the TODO so we keep track of it?